PhpAmqpLib/Channel/AbstractChannel.php

```diff
@@ -275,26 +276,33 @@ public function wait_content()
         return $msg;
     }
 
-    protected function build_msg_body($body_size) {
+    /**
+     * Reads and set the given message body by taking care of the possible message
+     * body size limit.
+     *
+     * @param \PhpAmqpLib\Message\AMQPMessage $message
+     */
+    protected function buildMessageBody($message)
+    {
         $body_parts = array();
         $body_received = 0;
-        $is_truncated = false;
-        while (bccomp($body_size, $body_received, 0) == 1) {
+
+        while (bccomp($message->getBodySize(), $body_received, 0) == 1) {
             list($frame_type, $payload) = $this->next_frame();
 
             $this->validate_body_frame($frame_type);
 
             $body_received = bcadd($body_received, mb_strlen($payload, 'ASCII'), 0);
 
-            if ( ! is_null($this->body_size_max) && $body_received > $this->body_size_max ) {
-                $is_truncated = true;
+            if (is_int($this->maxBodySize) && $body_received > $this->maxBodySize ) {
+                $message->setIsTruncated(true);
                 continue;
             }
 
             $body_parts[] = $payload;
         }
 
-        return array(implode('', $body_parts), $is_truncated);
+        $message->setBody(implode('', $body_parts));
```

Comment:
you see, from a functional programming point of view, I really don't like this style. Now the function is mutating it's arguments. It only works for messages, and so on. Before it could return you the body, you could do with it whatever you wanted.